### PR TITLE
kb layout independent keystroke info for tty backend also. see #1562

### PR DESCRIPTION
**Pull request by unxed**
also re-enables compact input mode

### CODE REVIEW

File: WinPort/src/Backend/TTY/TTYX/TTYX.cpp [hunk 1]
```diff
@@ -155,7 +155,26 @@ class TTYX
 			if ((cookie->evtype == XI_RawKeyPress || cookie->evtype == XI_RawKeyRelease) && cookie->data) {
 				const XIRawEvent *ev = (const XIRawEvent *)cookie->data;
 				//fprintf(stderr, "TTYXi: !!!!! %d %d\n", cookie->evtype == XI_RawKeyPress, ev->detail);
-				const KeySym ks = XkbKeycodeToKeysym(_display, ev->detail, 0 /*xkbState.group*/, 0 /*shift level*/);
+
+				XkbDescPtr xkb = XkbGetMap(_display, 0, XkbUseCoreKbd);
```
Comment (elfmz) [inline WinPort/src/Backend/TTY/TTYX/TTYX.cpp:159]: its better to call XkbGetMap/XkbGetControls/XkbGetNames in constructor of  TTYX remembering all needed things and do here only XkbKeycodeToKeysym

Comment (unxed) [reply WinPort/src/Backend/TTY/TTYX/TTYX.cpp:159]: If done this way, there may be errors if the user adds or removes keyboard layouts while far2l is already running.

Comment (elfmz) [reply WinPort/src/Backend/TTY/TTYX/TTYX.cpp:159]: I think its a very rare situation and user will restart far2l before posting bug here)
But once somebody will post a bug, will then use https://tronche.com/gui/x/xlib/events/window-state-change/mapping.html

Comment (unxed) [reply WinPort/src/Backend/TTY/TTYX/TTYX.cpp]: ok, fixed!

Comment (unxed) [reply WinPort/src/Backend/TTY/TTYX/TTYX.cpp]: Is there something else left to be done?

**Comment by unxed** [comment]
oh, there is a build problem in some configurations, probably fixed it in #1576

**Comment by unxed** [comment]
sorry, one more build problem, see #1577

**Pull request merged**